### PR TITLE
Brig offitsers

### PR DESCRIPTION
**Pull request by Kmc2000**
DISCLAIMER

Ported from:  https://github.com/yogstation13/yogstation/pull/385

THIS CODE WAS WRITTEN BY @Super3222  

I've adapted and butchered it heavily, super still did the original stuff

I've adapted it so that it works without jexp and added a casual state for the jumpsuit.

What this does:

Adds a "warden's bitch" role, effectively, they're there to lemon in the brig all round and do whatever the warden wants, they have sec officer weapons accesss and basic brig accesses to aid the warden

They have no server age limit, and are ideal for learning sec without the responsibilities of a normal officer.

Pictures:

Reasons for doing this

Even though jexp got murdered in cold blood by goof, i actually thought deputies were a fantastic idea, so i decided to port them, so we have the option of having them seperately from JEXP.

Deputies have now become pritson offitsers

Thanks go to super for originally making this

Fixed the fuckup, honk

🆑 Kmc2000
rscadd: Nanotrasen pshycho mark (tm) reports show that local wardens suffer from unmanageable stress, instead of passing security academy, candidates may now apply to assist station wardens as a deputy after 1 month's basic training.
/🆑

### CODE REVIEW

**Comment by TheSadFox** [comment]
My head hurts like fuck so pardon me for any typos/if this ends up as an incomprehensible mess.

This job is basically a gimped security officer that new players won't pick becouse all they want is "MOAR GEAR MORE ACCESS MOAR AUTHORITY", and experienced ones won't either becouse why would they, it only handicaps you and provides no different gameplay experience than that you would have as a normal officer.

Maybe instead make this job something like "wardens assistant"  or "prison officer", basically wardens buttboy, who's job would be to assist warden with processing prisoners, fetch shit for warden or do stuff for him outside of brig, ones that warden himself can't really do due to being forced to sit in brig almost all the time. Prison officer would have authority over normal officers inside brig simmilar to warden.

Having a buddy to to talk to during slow rounds, or help you handle shit during more chaotic rounds as a warden is amazing (i still can't forget those few rounds long, long time ago when Lan was my assistant, these were the very few rounds in which i didn't have suicidal thoughts as a warden), majority of your job being processing prisoners would help new players, if one decided to pick said job, learn the space law and punishments, and for more experienced players it would offer a different gameplay experience of a more mobile mini warden with a neat jumpsuit.

**Comment by Kmc2000** [comment]
Could potentially @SeikoTheSeal  but i'd need to look into how flavourtext works

Sounds like an interesting idea, but anyways sec officer is limited to 5 day account age on yogs, hence these guys.

As a side note, the warden and HOS could make them their buttboys anyways via IDS

**Comment by TheSadFox** [comment]
Timed whitelist is a joke.

**Comment by Alek2ander** [comment]
> ideal for learning sec without the responsibilities of a normal officer

but with all the powers of a sec officer

also, unimplanted and can be antag. with access to sec lockers.
basically crew should exterminate all the sec deputies for their own safety

**Comment by Time-Green** [comment]
Why do the deputies look more badass then  the actual officers?

**Comment by AsV9** [comment]
> Why do the deputies look more badass then the actual officers?

this. the uniform is a tad too fancy.

**Comment by Kmc2000** [comment]
> also, unimplanted and can be antag. with access to sec lockers.

Oops

**Comment by TheSadFox** [comment]
The uniform would totally fit a prison officer tho :^)

**Comment by george99g** [comment]
I'm going to side with Seiko on the prison officer thing.

**Comment by Kmc2000** [comment]
Alright i can mod him to be a prison offitser

HELP WANTED

I dunno how 2 flavourtext

**Comment by Kmc2000** [comment]
Updated with special snowflake flavour text

If these get merged, we'll need a wiki page doing for them, they still have the same access, as they're there to help triage and so they'll likely need officer gear to do so, all in all they answer directly to the warden and if he doesn't want them with gear, then they don't get to use gear.

**Comment by TheSadFox** [comment]
I love Krav Maga and i'd propose them to spawn with gloves, but then people would start yelling at me, so i won't.

**Comment by FluffySurvivor** [comment]
lesser officers??? 

give them a different role rather than nerfed officers

**Comment by TheSadFox** [comment]
> lesser officers??? 

Read comments, OP wasn't updated yet.

**Comment by Kmc2000** [comment]
Updated OP

**Comment by TheSadFox** [comment]
Might need a sec hud icon btw.

**Comment by Kmc2000** [comment]
> seiko

What do you think i am? some sort of coder or something?

I'll have a go

**Comment by Kmc2000** [comment]
Don't merge until i've:
Loyalty implanted them

Given them a sec icon

**Comment by EMT321** [comment]
Just add a securitron to armory a la Hippie/TG. Less likely to get robusted/harmbaton prisoners/waste peoples time.

Who would reasonably want this role?

**Comment by FluffySurvivor** [comment]
so this means 

we will see admins go apeshit because these deputies are not playing essentially warden?

**Comment by Kmc2000** [comment]
> Who would reasonably want this role?

Someone that wants to help out security in a way that doesnt involve getting memed on 5 seconds into the round?

**Comment by TheSadFox** [comment]
> Who would reasonably want this role?

Me.

**Comment by Stealthkibbler** [comment]
deputy would be a pretty chill job tbh

**Comment by AsV9** [comment]
OY BOT. FACK OFF.

**Comment by Kmc2000** [comment]
Only thing i can't work out now is how to add a sechud icon, ah well, if someone who knows the code better knows how do let me know

APART FROM THAT bwig offitsers are good to go

**Comment by FluffySurvivor** [comment]
*readies up gang/cult/rev buffs because +5 sec staff is broken af*

**Comment by Super3222** [comment]
Are you sure you want TWO prison officers? I mean having an assistant to hang around with wouldn't be so bad as a Warden but... two of them?

The amazing pros of this are you'll have someone to actually watch labour prisoners or perma prisoners. Cons are

File: code/modules/jobs/job_types/security.dm [hunk 1]
```diff
@@ -324,3 +324,54 @@ var/list/sec_departments = list("engineering", "supply", "medical", "science")
...
+ 	department_head = list("Head of Security")
+ 	department_flag = ENGSEC
+ 	faction = "Station"
+ 	total_positions = 2
+ 	spawn_positions = 2
+ 	supervisors = "the warden, you must help them with anything they need and assist with brig triage"
+ 	selection_color = "#ffeeee"
+ 	minimal_player_age = 0
+
+ 	outfit = /datum/outfit/job/deputy
```
Comment (Super3222) [inline code/modules/jobs/job_types/security.dm]: This is still named deputy.

File: code/modules/jobs/job_types/security.dm [hunk 2]
```diff
@@ -324,3 +324,54 @@ var/list/sec_departments = list("engineering", "supply", "medical", "science")
 /obj/item/device/radio/headset/headset_sec/alt/department/sci
 	keyslot = new /obj/item/device/encryptionkey/headset_sec
 	keyslot2 = new /obj/item/device/encryptionkey/headset_sci
+
+//Deputies, original code by super3222, adapted by Kmc2000//
+
+/datum/job/secdeputy
+ 	title = "Prison Officer"
+ 	flag = DEPUTY
```
Comment (Super3222) [inline code/modules/jobs/job_types/security.dm]: This is named deputy as well.

File: code/modules/jobs/jobs.dm [hunk 3]
```diff
@@ -9,8 +9,9 @@ var/const/OFFICER			=(1<<4)
 var/const/CHIEF				=(1<<5)
 var/const/ENGINEER			=(1<<6)
 var/const/ATMOSTECH			=(1<<7)
-var/const/AI				=(1<<8)
-var/const/CYBORG			=(1<<9)
+var/const/DEPUTY			=(1<<8)
```
Comment (Super3222) [inline code/modules/jobs/jobs.dm]: Err, I was planning on fixing this on my next commit. This should go under officer, and you'll need to reorder the numbers.

**Comment by Kmc2000** [comment]
Can lower it to one, discuss

File: code/modules/jobs/job_types/security.dm (hunk 2 above)
Comment (Kmc2000) [inline code/modules/jobs/job_types/security.dm]: Aware, was being lazy to avoid breaking everything

Comment (Super3222) [inline code/modules/jobs/job_types/security.dm]: As long as you change the correspondent constant variable in jobs.dm, you should be fine.

**Comment by Cruix** [comment]
blue_discus

**Comment by TheSadFox** [comment]
Cool with one. More isn't really needed. Also, consider adding a locker with a spare jumpsuit for them somewhere. Or just leave it as is for now and maybe once i start touching brig i'll add a small office for them somewhere.

**Comment by Kmc2000** [comment]
Changed spawns to one, cleaned up the CONST vars at the start and changed the types to brigofficer so succer can do his own depooties

**Comment by george99g** [comment]
This seems okay. I'll go ahead and merge it.

**Pull request merged**